xdem/spatial_tools.py

```diff
@@ -8,10 +8,20 @@
 
 import geoutils as gu
 import numpy as np
+import scipy
 import rasterio as rio
 import rasterio.warp
 from tqdm import tqdm
 
+try:
+    from sklearn.metrics import mean_squared_error, median_absolute_error
+    from sklearn.linear_model import (
+        LinearRegression, TheilSenRegressor, RANSACRegressor, HuberRegressor)
+    from sklearn.pipeline import make_pipeline
+    from sklearn.preprocessing import PolynomialFeatures
+    _has_sklearn = True
+except ImportError:
```

Comment:
Maybe too pedantic (you can ignore if you want) but this will be caught if a sub-dependency of sklearn is missing but sklearn exists. This should not happen with modern package managers, but it just caught my attention. Technically, one should also match the error message.

Reply:
Yes, I thought about the same thing (actually Fabien did!). I was quickly browsing to see if there's a good practice on how to handle optional imports. The only improvement I could find is to use importlib to check if a module exists or not: https://docs.python.org/3/library/importlib.html#importlib.abc.MetaPathFinder.find_spec

Reply:
So you could first check that sklearn was found. If yes, then import the submodules, if not, set _has_sklearn to False.
This could be made consistent in all the code.

Reply:
Opened #154